### ISSUE TITLE
Bug: Null pointer being thrown after parsing is done

### ISSUE THREAD

**Issue by ChitaAPC** [skadistats/clarity#257]
I am having a strange bug when parsing files.

I have some unit tests that are parsing a downloaded file using a LiveSource with a controllable runner. I wander if it is because the file i am using isn't actually a live recorded file (this doesn't happen when I use a MappedFileSource), but I thought I would bring it up anyway.

here is the stack for the error, i've included the unit test report to show how the exception is being thrown weirdly after the test is run and pass. 
I imagine I will probably need to give more context, but in the interest of not making a mile long issue, feel free to ask me if there is anything else that might be useful to figure this out.

```
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 50.065 sec
Exception in thread "clarity-livesource-watcher" java.lang.RuntimeException: java.lang.NullPointerException
	at skadistats.clarity.source.LiveSource.watcherThread(LiveSource.java:216)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at skadistats.clarity.source.LiveSource.handleFileChange(LiveSource.java:245)
	at skadistats.clarity.source.LiveSource.watcherThread(LiveSource.java:209)
	... 1 more
```
just another point, I have tried to wrap everything from my unit test to my entire main function in a try catch to debug what exactly is thrown the exception but nothing is being caught yet that exception is still being thrown.

I've also put a breakpoint within that line to try and catch the exception within the thread itself, but I think I am missing some knowledge of how clarity works under the hood, so that was not very helpful.

**Comment by spheenik** [comment]
Thanks for the report. 
Problem was on my side, if the file disappeared while the source was not stopped, the call of `file.capacity()` would throw an NPE.

Fixed with commit 86335cf0098b2258ee45a764f95634760b6338d2.

**Comment by spheenik** [comment]
just update to 2.6.2 to get the fix.
